crates/storage_models/src/payment_intent.rs

```diff
@@ -231,20 +232,25 @@ impl From<PaymentIntentUpdate> for PaymentIntentUpdateInternal {
                 amount_captured,
                 // customer_id,
                 return_url,
-                client_secret: make_client_secret_null_if_success(Some(status)),
+                client_secret: make_client_secret_null_based_on_status(Some(status)),
                 modified_at: Some(common_utils::date_time::now()),
                 ..Default::default()
             },
         }
     }
 }
 
-fn make_client_secret_null_if_success(
+fn make_client_secret_null_based_on_status(
     status: Option<storage_enums::IntentStatus>,
 ) -> Option<Option<String>> {
-    if status == Some(storage_enums::IntentStatus::Succeeded) {
-        Some(None)
-    } else {
-        None
-    }
+    status.and_then(|status| match status {
+        storage_enums::IntentStatus::Succeeded
+        | storage_enums::IntentStatus::Failed
+        | storage_enums::IntentStatus::Cancelled => Some(None),
```

Comment:
should be make this to bool instead of `Some(None)` and `None`?

Reply:
`Some(None)` deletes the value in the database and `None` leaves the value unedited. 

If we make return type bool we have to still write a wrapper function to convert it to Some(None) or None. 

Since `Some(None)` and `None` is specific to diesel we could have a generic wrapper function in generics file I guess.

Should I do it in this PR or take it up in a separate PR ?